examples/math/gsm8k/gsm8k_assertions.ipynb

```diff
@@ -0,0 +1,653 @@
+        "\n",
+        "Every call to the LM in a **DSPy** program needs to have a **Signature**.\n",
+        "\n",
+        "A signature consists of three simple elements:\n",
+        "\n",
+        "- A minimal description of the sub-task the LM is supposed to solve.\n",
+        "- A description of one or more input fields (e.g., input question) that we will give to the LM.\n",
+        "- A description of one or more output fields (e.g., the question's answer) that we will expect from the LM.\n",
+        "\n",
+        "Let's define a simple signature for basic math problem solving."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Ue7--j2Tk5qe"
+      },
+      "outputs": [],
+      "source": [
+        "class SimpleMathSignature(dspy.Signature):\n",
+        "    \"\"\"Answer the math question.\"\"\"\n",
+        "\n",
+        "    question = dspy.InputField(desc=\"A simple math question.\")\n",
+        "    answer = dspy.OutputField(desc=\"The answer to the math question.\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "P-lX8Grsl9wT"
+      },
+      "source": [
+        "In `SimpleMathSignature`, the docstring describes the sub-task here (i.e., answering math questions). Each `InputField` or `OutputField` can optionally contain a description `desc` too. When it's not given, it's inferred from the field's name (e.g., `question`).\n",
+        "\n",
+        "Notice that there isn't anything special about this signature in **DSPy**. We can just as easily define a signature that takes a long snippet from a PDF and outputs structured information, for instance.\n",
+        "\n",
+        "One trick for DSPy signature is that when it only contains simple fields performing straightforward tasks, we can replace the whole class definition with a syntactic sugar `question -> answer`. Now, lets define our first DSPy program with DSPy predictor. A predictor is a module that knows how to use the LM to implement a signature. Importantly, predictors can learn to fit their behavior to the task!"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "D3A1CH3foAFv"
+      },
+      "outputs": [],
+      "source": [
+        "basic_math_solver = dspy.Predict(\"question -> answer\") # Alternatively, we can write dspy.Predict(SimpleMathSignature)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "_GBP5jBUpRQP"
+      },
+      "source": [
+        "**`DSPy.Predict`** is the simplest DSPy predictor. Now, we can call this minimal _program_ with a hand crafted question:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "w1rZSbu2pQY3"
+      },
+      "outputs": [],
+      "source": [
+        "prediction = basic_math_solver(question=\"What is 1+1+1?\")\n",
```

Comment:
the cache seems to break for me here on just this query. the rest works however!